test/e2e/index.test.ts

```diff
@@ -1,6 +1,38 @@
 import { expect } from 'chai';
 import * as playwright from 'playwright';
 
+function sleep(timeoutMS: number): Promise<void> {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(), timeoutMS);
+  });
+}
+
+/**
+ * Attempts page.goto with retries
+ *
+ * @remarks The server and runner can be started up simultaneously
+ * @param page
+ * @param url
+ */
+async function attemptGoto(page: playwright.Page, url: string): Promise<boolean> {
+  const maxAttempts = 10;
```

Comment:
Why have a max attempt if we already have a timeout on the before task?

Reply:
To answer my own question, to get an accurate error message.

Reply:
Because the timeout does not necessarily translate to the number of attempts considering we await other async tasks before. The `maxAttempts` is just something to statically verify the loop doesn't run forever.